sql/core/src/test/scala/org/apache/spark/sql/streaming/MemorySinkSuite.scala

```diff
@@ -187,6 +187,31 @@ class MemorySinkSuite extends StreamTest with BeforeAndAfter {
     query.stop()
   }
 
+  test("MemoryPlan statistics for joining") {
```

Comment:
it would also be nicer to just implement a test on the MemoryPlan itself, without needing to start a stream, purely on statistics calculation